Form/Type/ColorSelectorType.php

```diff
@@ -33,7 +33,7 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults(array(
+        $defaults = array(
             'choices' => Colors::getAll(),
```

Comment:
There needs to be a new argument there. This argument should come from the a configurable container parameter.
In the unit tests, this argument should be set to false so that you don't have to mark all tests with the legacy annotation. The user should be able to do the following and not get any deprecation notice : 

```
sonata_core:
   my_parameter: false # defaults to true since defaulting to false would not be BC
```